src/clear/model/json_serializable.cr

```diff
@@ -0,0 +1,71 @@
+macro columns_to_instance_vars
+  struct Assigner
+    include JSON::Serializable
+
+    {% for name, settings in COLUMNS %}
+      getter {{name.id}} : {{settings[:type]}}?
+    {% end %}
+
+    def to_create
+      parse_from_json({{@type}}.new) # Loop through instance variables and assign to the newly created orm instance
+    end
+
+    def to_update(to_update_model)
+      parse_from_json(to_update_model) # Loop through instance variables and assign to the orm instance you are updating
+    end
+
+    macro finished
+      def parse_from_json(model)
+        {% for name, settings in COLUMNS %}
+          model.{{name.id}} = @{{name.id}}.not_nil! unless @{{name.id}}.nil?
+        {% end %}
+
+        model
+      end
+    end
+  end
+
+  def self.from_json(string_or_io : String | IO)
+    Assigner.from_json(string_or_io)
+  end
+
+  def self.new(string_or_io : String | IO)
+    Assigner.from_json(string_or_io).to_create
+  end
+
+  def self.create(string_or_io : String | IO)
+    self.new(string_or_io).save
+  end
+
+  def self.create!(string_or_io : String | IO)
+    self.new(string_or_io).save!
+  end
+
+  def set(string_or_io : String | IO)
+    Assigner.from_json(string_or_io).to_update(self)
+  end
+
+  def update(string_or_io : String | IO)
+    set(string_or_io).save
+  end
+
+  def update!(string_or_io : String | IO)
+    set(string_or_io).save!
+  end
```

Comment:
Are these new methods for clear?
If so, might be better to give them `_from_json` suffixes so it's explicit.
`String | IO` could be anything

Reply:
No these are existing methods for Clear, but they don't have an overload / the arguments do not allow String | IO.

These methods can be found here
https://github.com/anykeyh/clear/blob/5e233e549894010ed68ca949edf8f532d4fa086f/src/clear/model/modules/has_saving.cr

https://github.com/anykeyh/clear/blob/5e233e549894010ed68ca949edf8f532d4fa086f/src/clear/model/modules/class_methods.cr

Reply:
i'd not overload these with `String | IO`, as someone may want to add other serialization methods in the future (i.e. from yaml or even xml heaven forbid) and you'll have an issue. better to be explicit that you're parsing from json.

Reply:
ok cool I'll do that